scripts/spot-randomizer/functions/fn_onDamageTaken.sqf

```diff
@@ -0,0 +1,39 @@
+// CLIENT only!
+if (!hasInterface) exitWith {};
+
+// Only update for Zeus & Streamer
+if (isNull(getAssignedCuratorLogic player)) exitWith {};
+params ["_container", "_instigator", "_damage", "_ammo", "_damagePrv"];
+
+private _dmg_dealt = (_damage - _damagePrv) * 100; // This var is in %
+
+// Remember the last true damage dealing unit
+if !(isNull(_instigator)) then {
+    _container setVariable ["shootnscoot_lastDamageDealer", name _instigator];
+};
+
+// Special handling if target is destroyed. Update task & markers, then  cleanup EH + variables.
+// TODO:    The marker updates will not work when Zeus is JIP, as they only update when damage is dealt. Might have to do an "init" run on all target units for JIP, maybe in postInit?
+//          Must be extra cautious with initialization order, though!
+if ((playerSide in [civilian, independent]) && _damage >= 1) exitWith { // TODO: When exactly does ACE cookoff start? When should the target count as actually "destroyed"?
+    _station = (_container getVariable "shootnscoot_stationid");
+    _taskID = (str playerSide) + _station;
+    if ([_taskID] call BIS_fnc_taskExists) then { // task might not exist, e.g. for streamers
+        [_taskID, "FAILED"] call BIS_fnc_taskSetState;
+    };
+    _station setMarkerTypeLocal "mil_objective";
+    systemchat ([markerText _station, "has been destroyed by", _container getVariable "shootnscoot_lastDamageDealer"] joinString " ");
+    _station setMarkerTextLocal ((markerText _station) + " destroyed");
```

Comment:
@senshi-x I tested this on my machine 
- as a `Civilian` Zeus
- sitting in a BluFor MK6 mortar
- using it to kill a BluFor container
- all while hosting an MP session on my local machine
![image](https://github.com/gruppe-adler/Shoot_and_Scoot.Tanoa/assets/76476468/ae178a32-12eb-4fb8-a729-04618a77126e)

Never got any message or feedback on the map though. Feature does not appear to work. 🤔

Reply:
Feature works (as requested by you ;) ) only on non-servers. `remoteExec -2` prevents it from being run on servers. If you player-host (not on dedicated server), you are the server, so it will not trigger.

https://github.com/gruppe-adler/Shoot_and_Scoot.Tanoa/blob/2fa6dfe611d133c91cd31115529f8be0cab5f9c5/scripts/spot-randomizer/functions/fn_placeSupplyStation.sqf#L125
If you want it to work on player-hosted server client as well, I recommend reverting it to remoteexec everywhere (remove -2). This has near-zero performance impact anyway, because the called onDamageTaken does a `hasInterface` check as the first action, and remoteexec is triggered by the server in the first place (so no added network traffic).

Also recommend to start testing MP missions on dedicated servers with multiple local clients. For any second or more local client the damage handler would work even on player-hosted servers.

Reply:
> If you want it to work on player-hosted server client as well, I recommend reverting it to remoteexec everywhere (remove -2). This has near-zero performance impact anyway, because the called onDamageTaken does a `hasInterface` check as the first action, and remoteexec is triggered by the server in the first place (so no added network traffic).

I changed it to `0` and tested it like this. Works fine now in my test setup. 🙂 

> Also recommend to start testing MP missions on dedicated servers with multiple local clients. For any second or more local client the damage handler would work even on player-hosted servers.

Sure. I will organize such a test session.
For the time being I think you did everything necessary that I will now merge the PR into `main`.

Thanks for the awesome set of new features. I love it!! 🤩